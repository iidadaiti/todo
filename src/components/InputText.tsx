import '@spectrum-css/textfield'
import React, { ComponentProps, FC } from 'react'
import { mergeClassNames } from '../utils/classNames'

export interface InputTextProps
  extends Omit<ComponentProps<'input'>, 'children'> {
  type?: Extract<
    ComponentProps<'input'>['type'],
    | 'date'
    | 'datetime-local'
    | 'email'
    | 'month'
    | 'number'
    | 'password'
    | 'search'
    | 'tel'
    | 'text'
    | 'time'
    | 'url'
    | 'week'
  >
}

export const InputText: FC<InputTextProps> = ({
  className,
  type = 'text',
  ...props
}) => (
  <div className="spectrum-Textfield">
    <input
      type={type}
      className={mergeClassNames('spectrum-Textfield-input', className)}
      {...props}
    />
  </div>
)
