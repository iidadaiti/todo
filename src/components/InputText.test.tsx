import { render } from '@testing-library/react'
import React from 'react'
import { InputText } from './InputText'

test('InputText snapshot', () => {
  const result = render(<InputText />)
  expect(result.asFragment()).toMatchSnapshot()
})
