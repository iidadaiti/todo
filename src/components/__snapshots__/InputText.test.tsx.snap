// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText snapshot 1`] = `
<DocumentFragment>
  <div
    class="spectrum-Textfield"
  >
    <input
      class="spectrum-Textfield-input"
      type="text"
    />
  </div>
</DocumentFragment>
`;
